python/cudf/cudf/core/column/string.py

```diff
@@ -6172,11 +6204,15 @@ def to_pandas(
         nullable: bool = False,
         arrow_type: bool = False,
     ) -> pd.Index:
-        if nullable and not arrow_type:
-            pandas_array = pd.StringDtype().__from_arrow__(self.to_arrow())
+        if (
+            isinstance(self.dtype, pd.StringDtype)
```

Comment:
Should this still respect whether `nullable` or `arrow_type` was set?

Reply:
I'm thinkging we should just ignore `nullable` and `arrow_type` when pandas-compatibility mode is enabled and later when we have these types supported in non pandas compatibility mode aswell just drop `nullable` and `arrow_type` parameters, what do you say?

Reply:
I'm a little hesitant on dropping these parameters since these parameters are still pass down through `DataFrame/Index/Series.to_pandas`. So I think it's still valid for a cuDF classic user to specify e.g. `df.to_pandas(nullable=True)` and get a pandas objects with the pandas nullable extension types even if original data was not constructed with it.

Reply:
Sounds good, I preserved `nullable` and `arrow_type` behaviours in normal mode. The round-tripping will ignore these parameters only in pandas compatibility mode.